src/herder/HerderSCPDriver.cpp

```diff
@@ -1132,4 +1139,26 @@ HerderSCPDriver::wrapStellarValue(StellarValue const& sv)
     auto res = std::make_shared<SCPHerderValueWrapper>(sv, val, mHerder);
     return res;
 }
+
+void
+HerderSCPDriver::checkInvariants() const
+{
+    auto lcl = mApp.getLedgerManager().getLastClosedLedgerNum();
+    // Either tracking or last tracking must be set
+    // Tracking is ahead of or equal to LCL
+    if (mTrackingSCP)
+    {
+        releaseAssert(!mLastTrackingSCP);
+        releaseAssert(mTrackingSCP->mConsensusIndex >= lcl);
```

Comment:
this isn't true because captive core defers overlay, and resets Herder _after_ the initial catchup is done, so tracking ends up being the same as LCL. That being said, as per our discussion elsewhere, the general approach we might want to take here is to introduce a "booting" state in Herder, and only check the invariants when Herder is properly tracking the network.